### ISSUE TITLE
User Story dependencies

### ISSUE THREAD

**Issue by argentaiga** [taigaio/taiga-front#111]
Hi Taiga Team! I strongly believe that having a quick way (as a feature instead of a work around) to set dependencies between User Stories is fundamental for this type of tools. Often times there are series of stories that belong to an Epic that need to be implemented in certain order, so having an easy way to set the dependencies (which should then affect the order that these tasks are shown in the board, being Kanban or Scrum) is IMHO a "must have".

I'm not picky in the way it is implemented. It could be a simple tracker so the developer working on an issue quickly and obviously sees that the issue has a dependency, and then order the different boards (kanban or scrum) taking this in consideration. Or it could be done in a more complex way where one can set if the dependency is blocking or not, and many other bells and whistles. Up to you (although I tend to favor simple and solid over complex, which is why I like Taiga so much, but that is just my own preference).

I did a quick research and found that this issue was requested back in 2014 (and had lots of users supporting the request), but somehow it was closed. Here is the link to the old request: https://github.com/taigaio/taiga-back/issues/154

I'm creating this item by a suggestion by Pablo Ruiz, after I asked the question in the Taiga group. Thanks in advance for at least considering it! And keep up the fantastic job you guys do!

German P.

**Comment by muellert** [comment]
I would like to see the task titles mentioned, and possibly, the list of blocking task be sorted. Also, if the tasks blocking something all go into some resolved state, the ticket being blocked, should be (preferably) automatically unblocked. Maybe have two fields for blocking tasks, one with only the task numbers, and one with some prose, as some people seem to desire prose.

**Comment by wjl** [comment]
I really want to use Taiga, but it is basically impossible to manage a large project without a way to track and enforce dependencies between epics, stories, tasks, and issues. Bizarrely, this is a completely obvious and essential feature, yet is incredibly hard feature to find in modern project management tools.

For now most of my teams are still using _Bugzilla_ as an agile project management tool to track requirements, tasks, activities, as well as to manage sprints and milestones. It has a pretty ancient UI, but works surprisingly well. I would be **incredibly** happy to be able to switch interested teams over to Taiga, but it's not going to happen if we can't break down and track items with directional dependencies. Everyone loves the idea of Kanban boards, etc, but nobody wants to move to a tool that can't at the very least do the same basic things Bugzilla can do.

**Comment by staticfloat** [comment]
Is there a bug bounty that I could contribute to on this issue?  I'd be happy to pay a developer some small amount to prioritize this feature.

**Comment by marcin-kwiatkowski** [comment]
After using Jira for many years, I jumped into a project that uses Taiga. Relations between tasks is something that helped millions of times to order task (even between teams / projects) by their relations. In taiga I just can't do it properly, cannot block tasks by a particular user story from other project (ok, I know that blocking is possible but it's a plain text only).

Is there any change to implement relations between items? It is a standard feature that everyone just have to use.

**Comment by WhiskerLogic** [comment]
We're using Taiga now. It would be great to have dependencies between all issue types. Customizing the types of relationships would also be a plus.

**Comment by WhiskerLogic** [comment]
Relationships between all issue types would be great!

**Comment by mig5** [comment]
Somewhat related: being able to mark a story as blocked by another story (instead of the 'Blocker' being just free text) would be great!
